java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/DebugAdapterContext.java

```diff
@@ -116,6 +119,19 @@ public int getDebuggerLine(int clientLine) {
     void setClientPathsAreUri(boolean clientPathsAreUri) {
         this.clientPathsAreUri = clientPathsAreUri;
     }
+    
+    public boolean requestProcessTermination() {
+        if (processExecutorHandle != null) {
```

Comment:
Only correct if `requestProcessTermination` and `setProcessExecutorHandle` are always called by the same thread.

Reply:
I wouldn't count on that